### PR TITLE
FEATURE: Add Icon and Header to Off Canvas

### PR DESCRIPTION
**Pull request by deotte**
### Issue
#240 

### Problem
Currently, the `offCanvasComponent` does not support the ability to add a custom header to the top. The only way you can render the `offCanvasComponent` is with the default header having a large **close** button at the top. 

### Solution
Added another binding to the `goOffCanvasService.openOffCanvas` call to be able to pass in a `header`. Because of the interesting way that the `goOffCanvasComponent` is rendered from the `goOffCanvasService`, I could not directly add an input to the component itself.

### Before
Screen Shot 2019-10-09 at 3 09 14 PM

### After
Screen Shot 2019-10-24 at 3 45 25 PM

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.scss [hunk 1]
```diff
@@ -30,6 +30,14 @@
   align-self: flex-start;
   flex: 0 0 auto;
   width: 100%;
+  display: flex;
+  justify-content: space-between;
+  padding: 1rem;
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.scss]: Let's swap this value to be the `$column-gutter`

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html [hunk 2]
```diff
@@ -11,13 +11,8 @@
...
 >
   <div class="go-off-canvas__header">
-    <button
-      (click)="closeOffCanvas()"
-      class="go-off-canvas__button"
-      type="button"
-    >
-      Close
-    </button>
+    <h2>{{ header }}</h2>
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html]: @deotte what happens when someone doesn't pass in a header?

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div
   (click)="closeOffCanvas()"
   [@fade]="opened ? 'in' : 'out'"
-  class="go-off-canvas-backdrop"
+  class="go-column go-column--100 go-off-canvas-backdrop"
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html]: I'm not sure we need to apply these `go-column` classes here. Was this necessary?

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html [hunk 4]
```diff
@@ -11,13 +11,8 @@
...
   <div class="go-off-canvas__header">
-    <button
-      (click)="closeOffCanvas()"
-      class="go-off-canvas__button"
-      type="button"
-    >
-      Close
-    </button>
+    <h2>{{ header }}</h2>
+    <go-icon icon="clear" iconModifier="medium" iconClass="clickable" (click)="closeOffCanvas()"></go-icon>
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html]: @deotte we should actually switch this to be a `go-icon-button`. You may run into troubles with icon color and what not, but there is an issue open for that if you'd be interested in picking that one up as well. #263

File: projects/go-tester/src/app/components/off-canvas-test/off-canvas-test.component.ts [hunk 5]
```diff
@@ -1,9 +1,9 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
 
 @Component({
   selector: 'off-canvas-test',
   templateUrl: './off-canvas-test.component.html'
 })
 export class OffCanvasTestComponent {
-
+  @Input() header: string = '';
```
[the remark this answers is not in the document]
Comment (deotte) [reply projects/go-tester/src/app/components/off-canvas-test/off-canvas-test.component.ts]: Whoops. Yeah, that can be removed

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.module.ts [hunk 6]
```diff
@@ -1,16 +1,22 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
+import { BrowserModule } from '@angular/platform-browser';
```
[the remark this answers is not in the document]
Comment (deotte) [reply projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.module.ts]: Cool! Thanks.

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.scss [hunk 7]
```diff
@@ -30,6 +30,14 @@
...
+  align-items: center;
+}
+
+.go-off-canvas__header:last-child {
```
[the remark this answers is not in the document]
Comment (deotte) [reply projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.scss]: This is actually just a css trick I picked up while looking for a solution, within flexbox, to align the header to the left, and `go-icon` to the right in the example screenshot above. I could probably re-work this by doing a `align-self` or something in that nature.

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html (hunk 3 above)
Comment (deotte) [reply projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html]: This was probably just a habit from what I've been doing on our Angular conversions. I can remove that.

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html (hunk 2 above)
Comment (deotte) [reply projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.html]: You are right. I will change that over to an `*ngIf`.

File: projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, ViewChild, ComponentFactoryResolver } from '@angular/core';
+import { Component, ComponentFactoryResolver, Input, OnInit, ViewChild } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (deotte) [reply projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.ts]: Correct. I can remove that

File: projects/go-lib/src/styles/_placeholders.scss [hunk 9]
```diff
@@ -5,3 +5,7 @@
 %element--no-margin {
   margin: 0;
 }
+
+.clickable {
```
[the remark this answers is not in the document]
Comment (deotte) [reply projects/go-lib/src/styles/_placeholders.scss]: I think after switching this to the `go-icon-button`, I won't even need this anymore. I can remove it

**Pull request merged**